src/engine.c

```diff
@@ -125,6 +125,48 @@ ENGINE_readFile(ENGINE* engine, const char* path, size_t* lengthPtr, char** reas
   return readEntireFile(pathBuf, lengthPtr, reason);
 }
 
+internal bool
+ENGINE_fileExists(ENGINE* engine, const char* path) {
+ char pathBuf[PATH_MAX];
+
+ if (strncmp(path, "./", 2) == 0) {
+   strcpy(pathBuf, path + 2);
+ } else {
+   strcpy(pathBuf, path);
+ }
+ 
+ if (path[0] != '/') {
+   strcpy(pathBuf, BASEPATH_get());
+   strcat(pathBuf, path);
+ }
+ return doesFileExist(pathBuf);
+}
+
+internal int
+ENGINE_directoryExists(ENGINE* engine, const char* path) {
+ char pathBuf[PATH_MAX];
+
+ if (strncmp(path, "./", 2) == 0) {
+   strcpy(pathBuf, path + 2);
+ } else {
+   strcpy(pathBuf, path);
+ }
+ 
+ if (path[0] != '/') {
+   strcpy(pathBuf, BASEPATH_get());
+   strcat(pathBuf, path);
+ }
+ 
+ if (isDirectory(pathBuf)) {
+   return 1;
+ }
+ 
+ if (doesFileExist(pathBuf)) {
+   return 2;
+ }
+ return 0;
+}
```

Comment:
These two functions are very similar. If possible, we should merge the common code into a single function. `ENGINE_directoryExists` and `ENGINE_fileExists` could then call that common function, and interpret the results as appropriate :) 

Less duplication of code means less code that has to be maintained, and shipped to users.

Reply:
Replaced all that code with 2 function calls to functiosn in src/io.c